apps/src/p5lab/redux/animationList.js

```diff
@@ -236,7 +236,10 @@ function generateAnimationName(baseName, animationList) {
   // Match names with the form baseName_#
   for (let animation in animationList) {
     let animationName = animationList[animation].name;
-    if (animationName.substring(0, baseName.length) === baseName) {
+    if (
+      animationName &&
+      animationName.substring(0, baseName.length) === baseName
```

Comment:
nit: could this also be `animationName?.substring(...)`?

Reply:
done!

Reply:
Is there a tiny risk that this changes behaviour, such that if `animationName` is `null` or `undefined`, and `baseName` is `undefined`, then this conditional now succeeds, whereas before it didn't?

Reply:
Hmm, that's true...although `baseName.length` would also throw in that case?

Reply:
Oh good point.  Wasn't sure if it was actually possible with current code.

Reply:
I'll change it back, the point is not to throw in this situation